### DISCUSSION TITLE
Vue, Angular ou React? Qual lib/framewok você usaria para iniciar um projeto do zero?

### DISCUSSION THREAD

**Discussion by JimmyBastos** [frontendbr/forum#744] [Discussão]
Salve Galera do bem, como vão vocês?!!

Estou a realizar uma análise comparativa entre Vue, React e Angular. Foram definidos três cenários de aplicação para essas tecnologias: 

- _Single Page Applications_;
-  _Progressive Web Apps_;
- _Native Rendering_;

Após comparar essas ferramentas, será escolhida uma delas para desenvolver um protótipo simples em um desses cenários, mas é preciso da opinião de vocês, desenvolvedores, visto que o trabalho tem o objetivo de facilitar a escolha entre essas ferramentas perante as necessidades de mercado.

## **Qual dessas _libs/framewoks_ você usaria para iniciar um projeto do zero em cada um desses cenários, por quê?**
Observações:

- Você pode responder para quantos cenários quiser
- Estou completamente aberto a dicas e sugestões
- Qualquer duvida, seja livre para perguntar

Agradeço a todos desde já!

**Comment by doug2k1** [comment]
As três são maduras, já foram validadas em produção por grandes projetos, tem um ecossistema e comunidade ativos. Então acho que cai mais pro gosto pessoal ou habilidades que o time já tem.

Eu (se fosse pra tocar sozinho) usaria React porque me agrada mais e tenho mais afinidade.

**Comment by klarkc** [comment]
Se vc quer apostar no futuro e ao mesmo tempo estar seguro: Vuejs, pois possuem a especificação de web components no roadmap, por não ter uma barreira de entrada para novos devs (vanilla JS com separação de escopos  - HTML, JS e CSS) e documentação bem acessível. O ponto contra do Vuejs é que ainda não possui uma biblioteca de componentes já escritos  tão vasta quanto o Angular ou React.

Angular foi por um caminho q n gostei (typescript / compilação / complexidade) e React não lhe dá respostas, espera q vc resolva por si só colocando a mão na massa da arquitetura.

Esse é um resumo beemm resumido, e minha opinião, então cada um vai ter uma diferente na real.

**Comment by diessica** [comment]
> Se vc quer apostar no futuro e ao mesmo tempo estar seguro: Vuejs [...]

corrigindo: se você quer apostar no futuro e ao mesmo estar seguro, **javascript**. o resto é questão de opinião. 😉 

se você quer tomar uma decisão, eu sugiro
1) research séria em cima dos frameworks de seu interesse (artigos e stuff),
2) brincar um pouco com cada,
3) explorar documentação e comunidade.

acho aqui só vai rolar repetição superficial do que vários artigos sobre o assunto já falam de forma mais profunda e séria.

**Comment by JimmyBastos** [comment]
> 1. Research séria em cima dos _frameworks_ de seu interesse (artigos e stuff),
> 2. Brincar um pouco com cada,
> 3. Explorar documentação e comunidade.

Acredito então que estou no caminho certo. A ideia do trabalho é explanar o desenvolvimento Web com JavaScript, explorar os principais _framewoks_ disponíveis no mercado e seu potencial, além de sua aplicação em diferentes cenários.

**Comment by didiraja** [comment]
Algumas issues uteis #732 , #637 e essa falando de SEO #631

**Comment by JoseRFJuniorLLMs** [comment]
COMPILADOR QUE COMPILA TUDO PARA TUDO: O LLVM projeto de infraestrutura de compilador é uma "coleção de modulares e reutilizáveis compilador e toolchain tecnologias" usado para desenvolver compilador front-ends e back-ends .

LLVM é escrito em C ++ e é projetado para tempo de compilação , link-tempo , em tempo de execução e otimização de "idle-time" de programas escritos em arbitrários linguagens de programação . Originalmente implementado para C e C ++, o design agnóstico de linguagem do LLVM gerou uma grande variedade de front-ends : linguagens com compiladores que usam LLVM incluem ActionScript , Ada , C # , Common Lisp , Crystal , CUDA , D , Delphi, Fortran , Halide , Haskell , Java bytecode , Julia , Kotlin , Lua , Objetivo-C , Linguagem de sombreamento OpenGL , Pônei, Python , R , Ruby , Ferrugem , Scala , Swift e Xojo .
